spec/hydrogen-spec.coffee

```diff
@@ -106,94 +106,86 @@ describe 'Kernel manager', ->
-
-    describe 'getAllKernelSpecs', ->
-        it 'should return an array with specs', ->
-            atom.config.set 'Hydrogen.kernelspec', JSON.stringify kernelSpecs
-            allKernelSpecs = KernelManager::getAllKernelSpecs()
-
-            expect(allKernelSpecs.length).toEqual(2)
-            expect(allKernelSpecs[0]).toEqual(kernelSpecs.kernelspecs.ijavascript.spec)
-            expect(allKernelSpecs[1]).toEqual(kernelSpecs.kernelspecs.python2.spec)
+            expect(specs.length).toEqual(2)
+            expect(specs[0]).toEqual(firstKernelSpec.kernelspecs.ijavascript.spec)
+            expect(specs[1]).toEqual(secondKernelSpec.kernelspecs.python2.spec)
 
     describe 'getAllKernelSpecsFor', ->
         it 'should return an array with specs for given language', ->
-            atom.config.set 'Hydrogen.kernelspec', JSON.stringify kernelSpecs
-            allKernelSpecsForPython = KernelManager::getAllKernelSpecsFor('python')
+            @kernelManager.saveKernelSpecs kernelSpecsString
+            allKernelSpecsForPython = @kernelManager.getAllKernelSpecsFor('python')
 
             expect(allKernelSpecsForPython.length).toEqual(1)
             expect(allKernelSpecsForPython[0]).toEqual(kernelSpecs.kernelspecs.python2.spec)
 
         it 'should return an empty array', ->
-            atom.config.set 'Hydrogen.kernelspec', JSON.stringify kernelSpecs
-            allKernelSpecsForJulia = KernelManager::getAllKernelSpecsFor('julia')
+            @kernelManager.saveKernelSpecs kernelSpecsString
+            allKernelSpecsForJulia = @kernelManager.getAllKernelSpecsFor('julia')
 
             expect(allKernelSpecsForJulia).toEqual([])
 
     describe 'getKernelSpecFor', ->
         it 'should return spec for given language', ->
-            atom.config.set 'Hydrogen.kernelspec', JSON.stringify kernelSpecs
-            kernelSpecForPython = KernelManager::getKernelSpecFor('python')
+            @kernelManager.saveKernelSpecs kernelSpecsString
+            kernelSpecForPython = @kernelManager.getKernelSpecFor('python')
 
-            console.log kernelSpecForPython
             expect(kernelSpecForPython).toEqual(kernelSpecs.kernelspecs.python2.spec)
 
         it 'should return undefined', ->
-            atom.config.set 'Hydrogen.kernelspec', JSON.stringify kernelSpecs
-            kernelSpecForJulia = KernelManager::getKernelSpecFor('julia')
+            @kernelManager.saveKernelSpecs kernelSpecsString
+            kernelSpecForJulia = @kernelManager.getKernelSpecFor('julia')
 
             expect(kernelSpecForJulia).toBeUndefined()
 
     it 'should read lower case name from grammar', ->
         grammar = atom.grammars.getGrammars()[0]
-        expect(KernelManager::getLanguageFor grammar).toEqual('null grammar')
+        expect(@kernelManager.getLanguageFor grammar).toEqual('null grammar')
 
     it 'should update kernelspecs', ->
-        KernelManager::updateKernelSpecs()
+        @kernelManager.updateKernelSpecs()
```

Comment:
I think  we should add a callback to `updateKernelSpecs` rather than making it synchronous.
:) Javascript is all about asynchronous code!